### PR TITLE
Add Redis queue storage driver

### PR DESCRIPTION
**Pull request by SamJUK**
## Summary
- Stacked on #10 — adds a `Redis` implementation of `QueueStorageInterface` (`SADD`/atomic `RENAME`/`SMEMBERS`/`DEL`), giving native dedupe and the same atomic claim/drain guarantee as the DB driver, without touching MySQL.
- Connection resolves from a dedicated `cache_debounce.redis` env.php block, falling back to the existing `page_cache` Redis backend if absent (with a loud warning about that instance's eviction policy potentially dropping pending tags silently).
- Driver selection is a `Provider` decorator bound once in `di.xml`, switched via a scoped config value (`storage_driver`) that has no `system.xml` field — set only via `config:set`, never Admin-visible, no per-driver `di.xml` override module needed.
- CI: runs a Redis container alongside the Magento test container so the new integration tests have something to connect to.

## Test plan
- [x] `vendor/bin/phpunit Test/Unit` passes (verified against a real Magento install's classes, including the Credis mock-destructor edge case)
- [x] `vendor/bin/phpstan analyse` clean at the repo's configured level
- [ ] `dev/tests/integration` against the CI Redis service
- [ ] Manually confirm `config:set samjuk_cache_debounce/general/storage_driver redis` switches the driver and the field doesn't appear in Stores > Configuration

### CODE REVIEW

File: Test/Unit/Model/Storage/RedisTest.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    private $client;
+    private $connectionResolver;
+    private $storage;
+
+    protected function setUp(): void
+    {
+        $this->client = $this->getMockBuilder(\Credis_Client::class)
+            ->disableOriginalConstructor()
+            ->addMethods(['sAdd', 'rename', 'sMembers', 'del'])
+            ->getMock();
```
[the remark this answers is not in the document]
Comment (SamJUK) [reply Test/Unit/Model/Storage/RedisTest.php:24]: Claude: Not a real issue. Credis_Client dispatches sAdd/rename/sMembers/del via `__call` magic rather than declaring them as real methods, so `addMethods()` doesn't collide. CI is green across php74–php84, which confirms the mock builds fine. Fluff — skip.

File: Model/Storage/Redis/ConnectionResolver.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        $this->client = new \Credis_Client(
+            $config['host'],
+            (int)$config['port'],
+            null,
+            '',
+            (int)$config['database'],
+            $config['password'] ?: null
+        );
```
[the remark this answers is not in the document]
Comment (SamJUK) [reply Model/Storage/Redis/ConnectionResolver.php]: Claude: Confirmed, real gap. The fallback branch already defends missing keys with `??` defaults but the dedicated env.php branch doesn't — a config block missing e.g. `password` or `database` hits an undefined-index warning here. Cheap fix: mirror `resolveFallback()`'s null-coalescing pattern for the dedicated path.

File: etc/adminhtml/system.xml [hunk 3]
```diff
@@ -23,5 +23,26 @@
...
         </section>
+
+        <!--
+        Deliberately gated behind its own ACL resource (see etc/adminhtml/acl.xml),
+        not granted to any role by default — this is a deploy-time infra
+        decision, not something store admins should casually toggle. It only
+        exists so `bin/magento config:set` has a registered path to write to;
+        see README.md for the intended CLI-only workflow.
+        -->
+        <section id="samjuk_cache_debounce_advanced" translate="label" type="text" sortOrder="20" showInDefault="1" showInWebsite="0" showInStore="0">
```
[the remark this answers is not in the document]
Comment (SamJUK) [reply etc/adminhtml/system.xml]: Claude: Confirmed, real contradiction. `showInDefault="1"` plus an ACL resource gate doesn't actually hide this from Admin UI — the default Administrator role has every ACL resource granted, so it will show up there despite the README/comment claiming it's CLI-only and "in practice not reachable." Needs a decision: either genuinely hide the field or correct the doc claim.

File: Test/Integration/Model/Storage/RedisTest.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        // cleanup can't be broken by the same bug it's meant to guard against.
+        $this->rawClient = new \Credis_Client(
+            self::CONNECTION_CONFIG['host'],
+            (int)self::CONNECTION_CONFIG['port'],
+            null,
+            '',
+            (int)self::CONNECTION_CONFIG['database']
+        );
+        $this->rawClient->flushDb();
+    }
```
[the remark this answers is not in the document]
Comment (SamJUK) [reply Test/Integration/Model/Storage/RedisTest.php:36]: Claude: Confirmed, real risk, partially mitigated already — this uses Redis DB index 2 rather than 0, so it's not wiping the default DB. Still worth an explicit opt-in env var gate in case a dev's local Redis uses index 2 for something else.

File: Test/Integration/Model/Storage/Redis/ConnectionResolverTest.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    private $reader;
+
+    protected function setUp(): void
+    {
+        $this->reader = ObjectManager::getInstance()->get(DeploymentConfig\Reader::class);
+
+        (new \Credis_Client('127.0.0.1', 6379, null, '', self::DEDICATED_DATABASE))->flushDb();
+        (new \Credis_Client('127.0.0.1', 6379, null, '', self::FALLBACK_DATABASE))->flushDb();
+    }
```
[the remark this answers is not in the document]
Comment (SamJUK) [reply Test/Integration/Model/Storage/Redis/ConnectionResolverTest.php:46]: Claude: Same as the sibling RedisTest comment — real risk, same fix (opt-in env var gate before allowing destructive flushDb calls in this test).

**Comment by SamJUK** [comment]
Claude: Overall — good driver implementation, three things worth resolving before merge: (1) ConnectionResolver needs null-coalescing on the dedicated env.php config keys, same as the fallback branch already has; (2) the storage_driver Admin UI visibility contradicts the README's CLI-only claim — default Administrator role has full ACL access so it will show up, needs a decision either way; (3) both Redis integration test suites do destructive flushDb() with no opt-in gate — low risk since they use non-default DB indices, but worth gating explicitly. The MockBuilder addMethods comment is fluff, CI already proves it's fine.
